src/components/forms/Checkbox.jsx

```diff
@@ -1,78 +1,69 @@
-import React from 'react';
-import uuidv4 from 'uuid/v4';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
-import cx from 'classnames';
+import FDS from 'lib/dictionary/js/styleConstants';
 
 import CheckEmptyIcon from 'lib/icons/react/CheckEmptyIcon';
 import CheckFilledIcon from 'lib/icons/react/CheckFilledIcon';
 import CheckIndeterminateIcon from 'lib/icons/react/CheckIndeterminateIcon';
 
-/**
- * @param {Object} props react props
- * @returns {ReactElement}
- */
-const Checkbox = ({
-  label,
-  showLabel,
-  indeterminate,
-  disabled,
-  inputRef,
-  ...otherProps
-}) => {
-  const id = uuidv4();
+import { CustomCheckboxInput, CustomCheckboxContainer } from '@reach/checkbox';
 
-  const IconUnchecked = CheckEmptyIcon;
-  const IconChecked = indeterminate ? CheckIndeterminateIcon : CheckFilledIcon;
+const Checkbox = ({ label, defaultChecked, showLabel = true, ...props }) => {
+  const [checkedState, setChecked] = useState(props.checked || defaultChecked || false);
+  const checked = props.checked != null ? props.checked : checkedState;
+
+  // We should maybe fire onChange in "addition" to setChecked, based on whether we
+  // interpret checkbox as being "controlled" or not
+  const onChange = props.onChange ? props.onChange : (e) => setChecked(e.target.checked);
+
+  const CheckboxIcon = (() => {
+    if (checked) {
+      if (checked === 'mixed') {
+        return CheckIndeterminateIcon;
+      }
+      return CheckFilledIcon;
+    }
+    return CheckEmptyIcon;
+  })();
+
+  const Element = showLabel ? 'label' : 'div';
 
   return (
-    <div
-      className={cx('fdsCheckable fdsCheckbox', { 'fdsCheckable--disabled': disabled })}
-    >
-      <input
-        type="checkbox"
-        id={id}
-        className="media--xs"
-        disabled={disabled}
-        ref={inputRef}
-        {...otherProps}
-      />
-      <label className="flush--bottom" htmlFor={id}>
-        <span className="fdsCheckable-icon--checked">
-          <IconChecked size="xs" />
-        </span>
-        <span className="fdsCheckable-icon--unchecked">
-          <IconUnchecked size="xs" />
+    <Element className="display--inlineBlock">
+      <CustomCheckboxContainer
+        checked={props.checked != null ? props.checked : checked}
```

Comment:
This is where things get tricky - does the Reach component mandate that this input be fully controlled? If not, we should pass through `checked` or `defaultChecked` props instead of always setting `checked` from here